### PR TITLE
Search in OA list

### PR DESCRIPTION
**Pull request by Ha-Trinh**
### Release Note
Create Search in OA List page
### PR Check (for Reviewer)
- [x] "How to Confirm" is Correct

### Changes
Modify component search and OA list
### How to confirm
Confirm can search in OA list page

### CODE REVIEW

File: client/src/Components/Search.js [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
 import { Search } from 'semantic-ui-react'
 import _ from 'lodash'
 import '../styles/search.css'
 
-const handleSearchChange = (e, { value }) => {
-  console.log(value)
+const SearchEngine = (props) => {
+  return (
+    <div>
+      <Search className="search-bar"
```
Comment (hientuminh) [inline client/src/Components/Search.js]: Please add a placeholder for it
```
 "Search account, description, or business"
```

File: client/src/Components/Navigation.js [hunk 2]
```diff
@@ -1,16 +1,30 @@
...
     localStorage.removeItem('zola-user-token')
     removeSession()
   }
 
+  const officialAccounts = officialAccountsList.map(officialAccount => {
+    let newObjOfficialAccount = {}
+    newObjOfficialAccount['key'] = officialAccount.id
+    newObjOfficialAccount['value'] = officialAccount.id
+    newObjOfficialAccount['text'] = officialAccount.accountName
+    return newObjOfficialAccount
```
Comment (hientuminh) [inline client/src/Components/Navigation.js]: Please update the code. I think I don't need to create a new variable
```
return {
      key: officialAccount.id,
      value: officialAccount.id,
      text: officialAccount.accountName
    }
```

File: client/src/Components/Navigation.js [hunk 3]
```diff
@@ -26,8 +40,29 @@ const Navigation = ({ isSignedIn, removeSession }) => {
...
           <div className="navigation-content">
-            <Link to={'/official'}>Offical</Link>
-            <Link to={'/'} onClick={logOut}>Logout</Link>
+            {officialAccountsList.length > 0
+              ? (
+                <Dropdown
+                  options={officialAccounts}
+                  defaultValue={officialAccounts[0].value}
+                  onChange={routerOADetails}
+                  header=<div className="item header-nav"><Link to={'/official'}>Official Manager</Link></div>
```
Comment (hientuminh) [inline client/src/Components/Navigation.js]: I think we will move it to the dropdown menu of user.

**Comment by hientuminh** [comment]
@Ha-Trinh Plz update unit test

**Comment by hientuminh** [comment]
@Ha-Trinh Please change the base branch and resolve conflict.

File: client/src/Components/pages/Official.js [hunk 4]
```diff
@@ -15,14 +17,23 @@ const Official = () => {
...
+    })
+    setFilterOAList(filter)
+  }
+
   return (
     <div className='official'>
       <Container>
-        <SearchEngine></SearchEngine>
-        <TableBeauty result={officialAccountsList}></TableBeauty>
+        <SearchEngine handleSearchChange={handleSearchChange} result={filterOAList}></SearchEngine>
```
Comment (hientuminh) [inline client/src/Components/pages/Official.js]: Please remove `result={filterOAList}`. I think we don't need it

**Comment by hientuminh** [comment]
@Tu-LeVan Please do it as 2nd reviewer.

File: client/src/Components/Search.js [hunk 5]
```diff
@@ -1,18 +1,25 @@
 import React from 'react'
+import PropTypes from 'prop-types'
 import { Search } from 'semantic-ui-react'
 import _ from 'lodash'
 import '../styles/search.css'
 
-const handleSearchChange = (e, { value }) => {
-  console.log(value)
+const SearchEngine = ({handleSearchChange}) => {
```
Comment (Tu-LeVan) [inline client/src/Components/Search.js]: Add space before and after `handleSearchChange` props

File: client/src/Components/Search.js [hunk 6]
```diff
@@ -1,18 +1,25 @@
...
+    <div>
+      <Search className="search-bar"
+        onSearchChange= {_.debounce((e) => handleSearchChange(e.target.value), 500, {
+          leading: true,
+        })}
+        showNoResults = {false}
+        placeholder = "Search"
+      />
+    </div>
+  )
```
Comment (Tu-LeVan) [inline client/src/Components/Search.js]: Remove spaces before and after `=`

File: client/src/Components/pages/Official.js [hunk 7]
```diff
@@ -15,14 +17,23 @@ const Official = () => {
...
+        <TableBeauty result={filterOAList}></TableBeauty>
```
Comment (Tu-LeVan) [inline client/src/Components/pages/Official.js]: Update code to beautiful :smile: 
```suggestion
        <SearchEngine handleSearchChange={handleSearchChange} />
        <TableBeauty result={filterOAList} />
```

**Pull request merged**